### ISSUE TITLE
Resource.make and Concurrent.timeoutTo causes an infinite lock

### ISSUE THREAD

**Issue by iRevive** [monix/monix#1173]
#### The snippet
```scala
import cats.data.EitherT
import cats.effect.{Concurrent, Resource}
import monix.eval.Task
import monix.execution.Scheduler

import scala.concurrent.duration._

object Test {

  def main(args: Array[String]): Unit = {
    type F[A] = EitherT[Task, String, A]

    implicit val S: Scheduler = Scheduler.global
    implicit val F: Concurrent[F] = Concurrent.catsEitherTConcurrent

    val resource = Resource.make(F.delay[Unit](throw new RuntimeException("Boom")))(_ => F.unit)
    val fa = resource.use(_ => F.unit)
    val orElse = F.unit

    val result = Concurrent.timeoutTo(fa, 5.millis, orElse).value.attempt.runSyncUnsafe()

    println(s"Result $result")
  }

}
```

#### Output
The program hangs forever and the message is not printed.

#### Expected output

Printed in the console:
```
Left(java.lang.RuntimeException: Boom)
```

#### There are several situations when the program works fine:
1) With increased timeout (e.g. 500.millis)
```
val result = Concurrent.timeoutTo(fa, 500.millis, orElse).value.attempt.runSyncUnsafe()
```

2) With `Resource.liftF` instead of `Resource.make`
```
val resource = Resource.liftF(F.delay[Unit](throw new RuntimeException("Boom")))
```

3) Without `Concurrent.timeoutTo`
```
val result = fa.value.attempt.runSyncUnsafe()
```

4) With  `cats.effect.IO` instead of `monix.eval.Task`

#### Potential root cause

The program works fine with Monix 3.1.0.  
Perhaps such behavior caused by https://github.com/monix/monix/pull/1087.

**Comment by Avasil** [comment]
Thanks for the detailed report - I will investigate today and release `3.2.1` if necessary because we need to do it anyway due to ScalaJS 1.0.0 issue in browsers

**Comment by Avasil** [comment]
I have included the fix in `3.2.1`

**Comment by iRevive** [comment]
@Avasil awesome! Thank you for the quick fix.
